cpp/src/arrow/flight/sql/server_test.cc

```diff
@@ -184,8 +183,7 @@ class TestFlightSqlServer : public ::testing::Test {
   std::mutex server_ready_m;
 
   void RunServer() {
-    arrow::flight::Location location;
-    ARROW_CHECK_OK(arrow::flight::Location::ForGrpcTcp("localhost", port, &location));
+    Location location = *Location::ForGrpcTcp("localhost", port);
```

Comment:
nit, but this could be `ARROW_CHECK_OK(Location::ForGrpcTcp(...).Value(&location));` still to preserve the previous behavior (that said, it should really be `ASSERT_OK`)